core/plugins/org.polarsys.capella.core.projection.common/src/org/polarsys/capella/core/projection/common/TransitionHelper.java

```diff
@@ -489,54 +489,61 @@ public boolean isPropertyValueTransitionAvailable(EObject object) {
-    for (AbstractTrace trace : scenario.getIncomingTraces()) {
-      TraceableElement src = trace.getSourceElement();
-      if (src instanceof Scenario) {
-        Scenario existingScenario = (Scenario) src;
-        // Should be an Interface Scenario
-        if (existingScenario.getKind() == ScenarioKind.INTERFACE) {
-          BlockArchitecture architectureTarget = BlockArchitectureExt.getRootBlockArchitecture(existingScenario);
-          if ("SALA".equals(transition)) {
-            if (architectureSource instanceof SystemAnalysis && architectureTarget instanceof LogicalArchitecture) {
-              return false;
-            }
-          } else if ("LAPA".equals(transition)) {
-            if (architectureSource instanceof LogicalArchitecture
-                && architectureTarget instanceof PhysicalArchitecture) {
-              return false;
-            }
-          } else if ("PAEPBS".equals(transition)) {
-            if (architectureSource instanceof PhysicalArchitecture && architectureTarget instanceof EPBSArchitecture) {
-              return false;
+
+  private boolean isIS2ISTransitionAvailable(EObject object, String transition) {
+    if (object instanceof Scenario && ((Scenario) object).getKind() == ScenarioKind.INTERFACE) {
+      Scenario scenario = (Scenario) object;
+      BlockArchitecture architectureSource = BlockArchitectureExt.getRootBlockArchitecture(scenario);
+
+      // Disable command if there is a transitioned IS
+      for (AbstractTrace trace : scenario.getIncomingTraces()) {
+        TraceableElement src = trace.getSourceElement();
+        if (src instanceof Scenario) {
+          Scenario existingScenario = (Scenario) src;
+          // Should be an Interface Scenario
+          if (existingScenario.getKind() == ScenarioKind.INTERFACE) {
+            BlockArchitecture architectureTarget = BlockArchitectureExt.getRootBlockArchitecture(existingScenario);
+            if ("SALA".equals(transition)) {
+              if (architectureSource instanceof SystemAnalysis && architectureTarget instanceof LogicalArchitecture) {
+                return false;
+              }
+            } else if ("LAPA".equals(transition)) {
+              if (architectureSource instanceof LogicalArchitecture
+                  && architectureTarget instanceof PhysicalArchitecture) {
+                return false;
+              }
+            } else if ("PAEPBS".equals(transition)) {
+              if (architectureSource instanceof PhysicalArchitecture
+                  && architectureTarget instanceof EPBSArchitecture) {
+                return false;
+              }
             }
-          }
 
-          continue;
+            continue;
+          }
         }
       }
+      return true;
     }
-    return true;
+    return ((object instanceof SystemAnalysis) || (object instanceof LogicalArchitecture)
+        || (object instanceof PhysicalArchitecture))
+        || ((object instanceof CapellaElement)
```

Comment:
Is the check 
(object instanceof CapellaElement)
            && (CapellaLayerCheckingExt.isInContextLayer((CapellaElement) object)
                || CapellaLayerCheckingExt.isInLogicalLayer((CapellaElement) object)
                || CapellaLayerCheckingExt.isInPhysicalLayer((CapellaElement) object))
redundant (every object passed to the method is already sure to be a CapellaElement & belongs to SA/LA/PA)?

Reply:
Yes, it seems redundant, I will update.